### PR TITLE
Laboratory work #3, Kurnakova Irina - 20FPL1

### PR DESCRIPTION
**Pull request by IrinaKurnakova**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -222,7 +338,23 @@ def save(self, name: str) -> int:
...
+        new_key = ''
+        for trie in self.tries:
+            for key, value in trie.n_gram_frequencies.items():
+                for element in key:
+                    new_key += self.storage.get_letter_by_id(element)
+                freq_dict[new_key] = value
+                new_key = ''
+        profile_as_dict['n_words'] = self.n_words
+        profile_as_dict['name'] = self.language
+        with open('{}.json'.format(name), 'w', encoding="utf-8") as file:
```
Comment (WhiteJaeger) [inline lab_3/main.py:354]: `file` is the Python built-in. Consider using another name.

File: lab_3/main.py [hunk 2]
```diff
@@ -222,7 +338,23 @@ def save(self, name: str) -> int:
...
+            json_string = json.dumps(profile_as_dict)
+            file.write(json_string)
```
Comment (WhiteJaeger) [inline lab_3/main.py:356]: `json` has the `dump` method that combines these two lines.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py (hunk 1 above)
Comment (IrinaKurnakova) [reply lab_3/main.py:354]: Done

File: lab_3/main.py (hunk 2 above)
Comment (IrinaKurnakova) [reply lab_3/main.py:356]: Done

File: lab_3/main.py [hunk 3]
```diff
@@ -20,8 +21,41 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
-
+    if not isinstance(text, str):
+        return ()
+    invaluable_trash = ['`', '~', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+',
```
Comment (WhiteJaeger) [inline lab_3/main.py:26]: If the values of an array will not change, it is better to use `tuple`.

File: lab_3/main.py [hunk 4]
```diff
@@ -20,8 +21,41 @@ def tokenize_by_sentence(text: str) -> tuple:
...
+                        '=', '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    for symbol in invaluable_trash:
+        text = text.replace(symbol, '')
+    for key, value in {'ö': 'oe', 'ü': 'ue', 'ä': 'ae', 'ß': 'ss'}.items():
+        text = text.replace(key, value)
+    sentence = ''
+    all_sentences = []
+    for number, symbol in enumerate(text):
+        if symbol not in ['.', '?', '!', '…'] and number + 1 != len(text):
```
Comment (WhiteJaeger) [inline lab_3/main.py:36]: `['.', '?', '!', '…']` is better to define earlier, outside of the loop.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py (hunk 3 above)
Comment (IrinaKurnakova) [reply lab_3/main.py]: Done

File: lab_3/main.py (hunk 4 above)
Comment (IrinaKurnakova) [reply lab_3/main.py]: Done

**Pull request closed without merging**